packages/devextreme/js/__internal/ui/html_editor/ui/aiDialog.ts

```diff
@@ -143,11 +162,19 @@ export default class AIDialog extends BaseDialog<AIDialogResult> {
         this._currentCommand = e.value;
         this._commandOptionsList = this._commandsMap[e.value]?.options ?? [];
         this._currentOption = this._commandOptionsList?.[0];
-
-        this._isAskAICommandSelected = e.value === 'askAI';
+        this._isAskAICommandSelected = e.value === AI_DIALOG_ASKAI_COMMAND_NAME;
         this._askAIPrompt = '';
+        this._getCustomCommandPrompt = this._commandsMap[e.value]?.prompt;
 
         this._setDialogState(this._getInitialDialogState());
+
+        const shouldExecuteAICommand = !this._isAskAICommandSelected
+          && !this._isAICommandExecuted
```

Comment:
[excess code?]
SelectBoxes are disabled when request is executing. So looks like it's always `false`

And actually same for next `isOpen` check - how this value can be changed when popup is closed?

Reply:
isOpen: What about the case while the Popup is being opened? To keep the code execution consistent, I changed the code in the onValueChanged method - it should not run the command when the Popup is not open.

As for the use of _isAICommandExecuting - you are probably right. At least no test crashed 🤣🤣🤣

Reply:
[7f1723b3c5b4ba79e953ed1eececb465bd223080]